docker-image-src/5/coredb/docker-entrypoint.sh

```diff
@@ -424,6 +394,36 @@ if running_as_root; then
     find "${NEO4J_HOME}"/conf -type f -exec chmod -R 600 {} \;
 fi
 
+## == EXTRACT SECRETS FROM FILES ===
+# These environment variables are set by using docker secrets and they override their equivalent env vars
+# They are suffixed with _FILE and prefixed by the name of the env var they should override
+# e.g. NEO4J_AUTH_FILE will override the value of the NEO4J_AUTH
+# It's best to do this first so that the secrets are available for the rest of the script
+
+# Loop through all environment variables
+for variable_name in $(printenv | awk -F= '{print $1}'); do
+  # Check if the variable ends with "_FILE"
+  if [[ $variable_name == "*_FILE" ]]; then
+    # Create a new variable name by removing the "_FILE" suffix
+    base_variable_name=${variable_name%_FILE}
+
+    # Get the value of the _FILE variable
+    secret_file_path="${!variable_name}"
+
+    # Check if the file exists and is readable, then read its contents
+    if [[ -f "$secret_file_path" ]]; then
+      secret_value=$(<"$secret_file_path")
+    else
+      # If it's not a file error
+      echo >&2 "The secret file '$secret_file_path' does not exist or is not readable. Make sure you have correctly configured docker secrets."
+      exit 1
+    fi
```

Comment:
nice error behaviour! Could you make a test for it :slightly_smiling_face: 
So the conditions are:
1. secrets file does not exist
2. secrets file exists but is not readable. This could be hard to reproduce though.

Also it just occurs to me, that if there is a non-secret env with a name that ends in `_FILE` then neo4j will fail to start (or at least there would be weird behaviour). For example if someone decides have a variable like `BACKUP_FILE` pointing to a huge dump file
Maybe there needs to be a check that the secret_file_path is actually under /run/secrets before we read it into the env?

Reply:
This didn't make sense, because container won't reach that point if file doesn't exist or is not readable. Docker handles that

Reply:
This was the original code:
```
    if [[ -f "$secret_file_path" ]]; then
      secret_value=$(<"$secret_file_path")
    else
      # If it's not a file error
      echo >&2 "The secret file '$secret_file_path' does not exist or is not readable. Make sure you have correctly configured docker secrets."
      exit 1
    fi
```

If the file does not exist, it errors nicely.
If the file is not readable, then the container will fail messily. Having a nice error message to explain the failure is a good thing. 
There's already an `is_readable` function in the entrypoint you could use. Just skipping the check altogether so that docker just fails is a worse solution than before.